app/src/main/java/mozilla/fenix/MainActivity.kt

```diff
@@ -17,26 +17,38 @@ import mozilla.fenix.components.FeatureLifecycleObserver
 import mozilla.fenix.ext.components
 
 class MainActivity : AppCompatActivity() {
+    private var toolbarFeature: ToolbarFeature? = null
+    private var sessionFeature: SessionFeature? = null
```

Comment:
Instead of making this a nullable type we could make use of Kotlin's `lateinit` feature here:
https://kotlinlang.org/docs/reference/properties.html

In an activity it's pretty much standard that nothing happens before onCreate() is run. So I think we guarantee that those properties will never be actually `null`.